addon/services/infinity.js

```diff
@@ -1,35 +1,58 @@
 import Service from '@ember/service';
 import InfinityModel from 'ember-infinity/lib/infinity-model';
+import InfinityPromiseArray from 'ember-infinity/lib/infinity-promise-array';
+import BoundParamsMixin from 'ember-infinity/mixins/bound-params';
 import EmberError from '@ember/error';
-import { isEmpty } from '@ember/utils';
+import { A } from '@ember/array';
+import { isEmpty, typeOf } from '@ember/utils';
 import { scheduleOnce } from '@ember/runloop';
 import { get, set } from '@ember/object';
-
-let checkInstanceOf = (infinityModel) => {
-  if (!(infinityModel instanceof InfinityModel)) {
-    throw new EmberError("Ember Infinity: You must pass an Infinity Model instance as the first argument");
+import { checkInstanceOf, convertToArray, findElem, objectAssign, paramsCheck } from '../utils';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+/**
+ * { 'products': { future_timestamp: infinityModel } }
+ * contains an array of Array Proxies
+ * only called when need to re-cache the collection
+ *
+ * @method cacheInfinityCollection
+ * @param Ember.Array _cachedCollection
+ * @param Object infinityModel
+ * @param String identifier
+ * @param String timestamp
+ * @return Object
+ */
+let cacheInfinityCollection = (_cachedCollection, infinityModel, identifier, timestamp) => {
+  if (_cachedCollection && _cachedCollection[identifier]) {
+    // 1. first clear out elements from object since we are expired
+    _cachedCollection[identifier] = {};
   }
-  return true;
-};
 
-let convertToArray = (queryObject) => {
-  if (queryObject.toArray) {
-    return queryObject.toArray();
-  }
-  return queryObject;
+  // 2. Set new timestamp for identifier
+  let future_timestamp = Date.now() + timestamp;
```

Comment:
Can we do `new Date().getTime()` instead of `Date.now()` for sliiiightly better compat?